tinygrad/tensor.py

```diff
@@ -281,7 +281,7 @@ def data(self) -> memoryview:
     assert self.dtype.fmt is not None, f"no fmt dtype for {self.dtype}"
     assert all_int(self.shape), f"no data if shape is symbolic, {self.shape=}"
     if TYPE_CHECKING or sys.version_info < (3, 12): assert self.dtype.fmt != "e"
-    return self._data().cast(self.dtype.fmt) if 0 in self.shape else self._data().cast(self.dtype.fmt, self.shape)
+    return cast(memoryview, self._data().cast(self.dtype.fmt) if 0 in self.shape else self._data().cast(self.dtype.fmt, self.shape))
```

Comment:
Cast `memoryview[int] | memoryview[float] | memoryview[bool]` to `memoryview`.

Reply:
Any reason to not make return type `memoryview[int] | memoryview[float] | memoryview[bool]`?

Reply:
In the [typeshed stub](https://github.com/python/mypy/blob/v1.13.0/mypy/typeshed/stdlib/builtins.pyi#L739), `memoryview` is generically defined as:

```python
_I = TypeVar("_I", default=int)  
class memoryview(Sequence[_I]):
  ...

```
Which leads to the typechecker substituting every `memoryview` type decl with `memoryview[int]`.

Changing the return type will require to change the types of a lot of variables/function args from `memoryview` to the broader `memoryview[int] | memoryview[float] | memoryview[bool]`. For example all the `Allocator._copyin` implementations (and so on).

I can use a type alias such as `MemoryviewType = memoryview[int] | memoryview[float] | memoryview[bool]` but that will be very confusing with both `MemoryviewType` and `memoryview` scattered across the codebase.

Imo the cleanest solution is to return a `memoryview` or a `memoryview[Any]`

Reply:
kk, cast is fine